pkg/operator/resources/asyncapi/status.go

```diff
@@ -297,6 +297,7 @@ func addPodToReplicaCounts(pod *kcore.Pod, deployment *kapps.Deployment, counts
 }
 
 func isPodSpecLatest(deployment *kapps.Deployment, pod *kcore.Pod) bool {
```

Comment:
note to reviewer: this is being called with the api deployment and the gateway deployment. I believe the gateway deployment can never be out of date, so therefore it's ok that these labels don't exist on the gateway deployment (which means this will always return `true`).

Reply:
That makes sense.